### PR TITLE
bugfix/java pipeline

### PR DESCRIPTION
**Pull request by baywet**
this pull requests makes the java pipeline production ready by:
- adding known issues for cases that are not supported by the java SDK
- adding java15 installation in the pipeline
- adding missing imports for unit test infrastructure

Related https://github.com/microsoftgraph/microsoft-graph-explorer-api/pull/363

### CODE REVIEW

File: azure-pipelines/java-prepare-template.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+steps:
```
Comment (zengin) [inline azure-pipelines/java-prepare-template.yml:1]: @MIchaelMainer once mentioned that there is a way to make this part of the agents we are using, but I don't remember the details. We should be good with this explicit step for now.

**Review by zengin — APPROVED** [review]

File: azure-pipelines/java-prepare-template.yml (hunk 1 above)
Comment (baywet) [reply azure-pipelines/java-prepare-template.yml:1]: the alternative method I know would be to add the java version as a build requirement and install the SDK on our own hosted agent. but that'd require maintenance over time.
We could use containers, but I haven't spent much time looking into containers with java + dotnet.
Maybe there's another way I don't know about though?

Comment (zengin) [reply azure-pipelines/java-prepare-template.yml:1]: I found what I was referring to. https://github.com/actions/virtual-environments/pull/588

There is a way to modify Azure virtual environments with direct PRs, but I am not sure if this particular change is generic enough to apply for everyone.

**Comment by baywet** [comment]
@zengin this is ready for final review.
You can see the checks are canceled, that's because I ran those instead with a branch contains all the latest snippet generation changes:
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31267&view=ms.vss-test-web.build-test-results-tab&runId=1005926&resultId=100005&paneView=debug
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31268&view=ms.vss-test-web.build-test-results-tab
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31269&view=ms.vss-test-web.build-test-results-tab
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31270&view=ms.vss-test-web.build-test-results-tab
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31272&view=ms.vss-test-web.build-test-results-tab
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31274&view=ms.vss-test-web.build-test-results-tab
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31271&view=ms.vss-test-web.build-test-results-tab
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31273&view=ms.vss-test-web.build-test-results-tab

- all the known issues pipelines only have failing tests
- all the build pipelines only have passing tests. Except for the 2 java ones, this is due to a flaky test because of JVM/compile environment warmup for which I just committed a fix.

and while this is still running, these builds don't have the warmup issue anymore thanks to the fix
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31299&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=006e6f7c-8314-5e02-a3df-483f3a1f4cfa
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31300&view=results

**Review by zengin — APPROVED** [review]

**Pull request merged**